### PR TITLE
#162170730 User should be able to comment on a highlighted text

### PR DESCRIPTION
**Pull request by okoroemeka**
#### What does this PR do?
Have functionality for commenting on a highlighted text in an article

#### Description of Task to be completed?
Have the following endpoint working
`POST api/articles/:slug/comments/highLightText`

#### How should this be manually tested?
Clone the repo, cd into it then run `npm install`. Use the `env.sample` file to set the required env variables
* Use Postman to test the endpoint

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170730](https://www.pivotaltracker.com/story/show/162170730)

#### Screenshots (If applicable)
screen shot 2018-12-14 at 7 21 56 am

#### Questions:
N/A

### CODE REVIEW

File: src/controllers/HighLightedText.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import errorResponse from '../helpers';
+
+const { HighlightedText: HighLightedTextModel, Article } = models;
+/**
+ * @description class representing comment on Highlightedtext Controller
+ * @class HighLightedText
+ */
+class HighLightedText {
+  /**
+   * @description - This method is responsible for creating new articles
```
Comment (oluwajuwon) [inline src/controllers/HighLightedText.js]: hi @okoroemeka your description says this method is for creating new articles, is that right?

Comment (okoroemeka) [reply src/controllers/HighLightedText.js]: Nope. Thanks for pointing that out. I will fix it now.

Comment (okoroemeka) [reply src/controllers/HighLightedText.js]: Fixed

File: src/controllers/HighLightedText.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+   * @memberof HighLightedText
+   */
+  static async createComment(request, response) {
+    const articleSlug = request.params.slug;
+    const userId = request.userData.payload.id;
+    const { text, comment } = request.body;
+    try {
+      const article = await Article.findOne({
+        where: {
+          slug: articleSlug
```
Comment (frostyblok) [inline src/controllers/HighLightedText.js]: Hi Emeka,  nice work here, but I think you could have reduced the noise here by destructuring the 'articleSlug' in line 19. You could have done something like...

const { slug } = request.params.slug;

Then line 25 becomes, ...where: { slug }.

I would like to know what you think, though.

File: src/controllers/HighLightedText.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      };
+      const createdComment = await HighLightedTextModel.create(payLoad);
+      return response.status(201).json({
+        status: 'Success',
+        message: 'Comment created successfully',
+        comment: createdComment
+      });
+    } catch (error) {
+      response.status(500).json({
+        status: 'Error',
```
Comment (frostyblok) [inline src/controllers/HighLightedText.js]: I like the creativity here, but you might need to stick with the team's convention.

**Review by frostyblok — COMMENTED** [review]
Hi, I am motivated about how you were able to complete this task. I like your clean implementation. I  have left some comments I would love you to look into for changes. If you would love to keep them as they are, I would love to know your reasons, so I can learn from them. Thank you!

File: src/controllers/HighLightedText.js (hunk 3 above)
Comment (okoroemeka) [reply src/controllers/HighLightedText.js]: Thanks for pointing that out. It has been resolved.

File: src/controllers/HighLightedText.js (hunk 2 above)
Comment (okoroemeka) [reply src/controllers/HighLightedText.js]: That's true. It has been fixed.

**Review by frostyblok — APPROVED** [review]

**Review by luqmanoop — APPROVED** [review]
Job well-done @okoroemeka

**Comment by omokaro-faith** [comment]
@okoroemeka Please resolve conflicts.

**Pull request merged**